key_management/src/wallet.rs

```diff
@@ -0,0 +1,181 @@
+        let k = Key::new(key_info)?;
+        let mut owned_string = "wallet-".to_owned();
+        owned_string.push_str(k.address.to_string().as_ref());
+        self.keystore.put(owned_string, k.key_info)?;
+        Ok(k.address)
+    }
+
+    /// Return a Vec that contains all of the Addresses in the Wallet's KeyStore
+    pub fn list_addrs(&self) -> Result<Vec<Address>, Error> {
+        let mut all = self.keystore.list();
+        all.sort();
+        let mut out = Vec::new();
+        for i in all {
+            if i.starts_with("wallet-") {
+                // TODO replace this with strip_prefix after it has been added to stable rust
+                let name = i.trim_start_matches("wallet-");
+                let addr = Address::from_str(name).map_err(|err| Error::Other(err.to_string()))?;
+                out.push(addr);
+            }
+        }
+        Ok(out)
+    }
+
+    /// Return the Address of the default KeyInfo in the Wallet
+    pub fn get_default(&self) -> Result<Address, Error> {
+        let key_info = self.keystore.get(&"default".to_string())?;
+        let k = Key::new(key_info)?;
+        Ok(k.address)
+    }
+
+    /// Set a default KeyInfo to the Wallet
+    pub fn set_default(&mut self, addr: Address) -> Result<(), Error> {
+        let mut owned_string = "wallet-".to_owned();
+        owned_string.push_str(addr.to_string().as_ref());
+        let key_info = self.keystore.get(&owned_string)?.clone();
+        // TODO change this code to not exit if there is no kv pair with default key in keystore
+        self.keystore.remove("default".to_string())?; // This line should unregister current default key then continue
+        self.keystore.put("default".to_string(), key_info)?;
+        Ok(())
+    }
+
+    /// Generate a new Key that fits the requirement of the given SignatureType
+    pub fn generate_key(&mut self, typ: SignatureType) -> Result<Address, Error> {
+        let key = generate_key(typ)?;
+        let mut owned_string = "wallet-".to_owned();
+        owned_string.push_str(key.address.to_string().as_ref());
+        self.keystore.put(owned_string, key.key_info.clone())?;
+        self.keys.insert(key.address, key.clone());
+        let value = self.keystore.get(&"default".to_string());
+        if value.is_err() {
+            self.keystore
+                .put("default".to_string(), key.key_info.clone())
+                .map_err(|err| Error::Other(err.to_string()))?;
+        }
+
+        Ok(key.address)
+    }
+
+    /// Return whether or not the Wallet contains a Key that is resolved by the supplied Address
+    pub fn has_key(&mut self, addr: &Address) -> bool {
+        self.find_key(addr).map_or_else(|_| false, |_| true)
+    }
+}
+
+/// Return the String that corresponds to each Signature type
+pub fn kstore_sig_type(typ: SignatureType) -> String {
+    match typ {
+        SignatureType::Secp256 => "secp256k1".to_string(),
+        _ => "bls".to_string(),
```

Comment:
I don't think you need this at all, but if you do, just make it explicit (could run into potentially an issue if other types are added in future)
```suggestion
        SignatureType::BLS => "bls".to_string(),
```